### PR TITLE
Use batch mode in extensions

### PR DESCRIPTION
**Pull request by timio23**
A very simple change which allows batchMode and batchActive to be used in extensions. I've been running this in production for 3 months now and can confirm this is working with no impact on existing features.

This allows us to adjust an interface during batch mode such as limiting actions or show a presentation notice

### CODE REVIEW

**Comment by timio23** [comment]
Hi @rijkvanzanten, would you like me to create an issue for this? I wouldn't consider this a bug. Here is the original Feature Request: #17553

**Comment by rijkvanzanten** [comment]
@timio23 Heya! We just revised our feature request workflow to make sure that PRs can be reviewed/merged efficiently in [our contributing guidelines](https://docs.directus.io/contributing/introduction.html). As per this workflow, we should technically close this PR as it doesn't follow the contributing guidelines. That being said, seeing this workflow was put together after your feature request was already live, and seeing it's such a tiny improvement, I'll accept it as-is. Please do be aware of that contributing guideline for other new features moving forwards though 🙂

**Comment by rijkvanzanten** [comment]
That being said.. I don't have "maintainer can edit this PR" access, so can you make sure the linter warnings are resolved and the changeset is created? Thanks! 🙌🏻

**Comment by timio23** [comment]
Thanks @rijkvanzanten, the changeset has been created (rude-bees-exercise.md 🤣) and lint warnings addressed

**Review by paescuj — APPROVED** [review]

File: .changeset/rude-bees-exercise.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@directus/app": patch
+---
+
+Allow batchMode and batchActive to be used in interface extensions
```
Comment (paescuj) [inline .changeset/rude-bees-exercise.md:5]: We use past tense for changsets 👍 
```suggestion
Allowed batchMode and batchActive to be used in interface extensions
```

Comment (rijkvanzanten) [reply .changeset/rude-bees-exercise.md:5]: I'll fix that on main so we don't have to delay the merge/launch 👍🏻

**Review by rijkvanzanten — APPROVED** [review]

File: .changeset/rude-bees-exercise.md (hunk 1 above)
Comment (rijkvanzanten) [reply .changeset/rude-bees-exercise.md:5]: @paescuj On second thought, I think this is a case where present tense makes more sense! "Allowed to do xyz" sounds like you could do it, but not anymore, whereas "Allow xyz" feels like "We started allowing you do to xyz" 🤔 

Bug fixes are different though, as "Fix issue 123" sounds like something you're about to do, whereas "Fixed 123" sounds like it's done and included 🙂

**Comment by timio23** [comment]
Exciting times. Thanks @rijkvanzanten

File: .changeset/rude-bees-exercise.md (hunk 1 above)
Comment (paescuj) [reply .changeset/rude-bees-exercise.md:5]: Haha yeah agreed 100% 😄

**Comment by chetank628** [comment]
can you explain a little bit more about this sorry i am not clear about this what this does..?

**Comment by timio23** [comment]
Hi @chetank628, there are 2 properties in Directus that control the batch edit window called batchMode and batchActive. This change makes these properties available to interfaces. As it stands, this changes nothing in Directus.

**batchMode** is `true` when the Batch Edit window is open, `false` when closed
**batchActive** is `true` when a field is "ticked" for editing and `false` if unticked.

What I've done is created an interface extension that uses the batchMode to see if the Batch Edit window is active and presents limited Status options to the user.

For my extension, you can create choices to present to the user. When creating the choices, I have a checkbox to allow that choice in the batch editor:
image

Then in my interface.vue file, I include the batchMode and batchActive properties (although I didn't need the batchActive in the end):

```
props: {
    batchMode: {
        type: Boolean,
        default: false,
    },
    batchActive: {
        type: Boolean,
        default: false,
    },
}
```

And use the batchMode in a filter on the v-for loop that renders my status options:

```
<button 
    v-for="item in choices.filter(
        choice => (batchMode === true && choice.allow_batch === true ) || (batchMode === false)
    )"
></button>
```

The result being this in batch edit:
image

And this when editing a single record:
image

Something like this could also be linked to permission roles as well.

**Pull request merged**